### PR TITLE
Schema overview page, and readme TOC links.

### PR DESCRIPTION
**Pull request by warpfork**
Riffing somewhat heavily from what's currently incubating in the other
repo at https://github.com/ipld/go-ipld-prime/blob/master/doc/schema.md

Further iteration on this will be super welcome; it's a bit abrupt.

### CODE REVIEW

**Review by vmx — APPROVED** [review]
Sweet!

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:8]: the relationship between "type system" and "schemas" seems to be in debate at the moment, we need to have a clear resolution of these two ideas before this lands

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:31]: we've so far avoided using "array" and stuck to "list" I think, maybe it'd be safer to keep it that way, or at least put "list" primary, either "typed lists" or "typed lists / arrays"

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:32]: should we reinforce "kinds" here too? "primitives / kinds" or  "primitives / IPLD Data Model kinds"

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:47]: ahh, "to use in place", a question I've been meaning to raise about aliases is: does a serialiser use the primary or the alias or is it left up to the implementation because it doesn't matter? is the purpose of an alias to allow the _deserialising_ of data with an alternative form or to allow the _serialising_ of data in an alternate (maybe more efficient?) form, or some undefined combination of both? herein lies both the consistent-hash problem and the consistent-across-implementations problem.

doesn't need to be addressed here, just noting for future discussion

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:49]: you're using "kind" here so I think that suggests you should use it with or instead of "primitive" above (although I quite like using it _with_ "primitive" as it helps explain "kind" which is bound to be unfamiliar to some audience)

File: schema/overview.md
Comment (rvagg) [inline schema/overview.md:56]: in debate/flux, needs to be resolved before this can land

File: schema/overview.md
Comment (warpfork) [reply schema/overview.md:47]: **must** use alias.  consistency problem is right -- both for hashing, and also for cardinality correctness even aside from hashing.

File: schema/overview.md
Comment (warpfork) [reply schema/overview.md:8]: I've been using the two more or less interchangeably.  "IPLD Schemas" are the definite noun (we're *not* calling this the proper-noun "IPLD type system")... but they're a "type system" of some general sort.

I think we hit a "limits of language" thing somewhere in this area, tbh.  Lots of this system's designs are coming from "type theory" (for programming languages) -- so the term seems to apply -- even though we're explicitly bending and limiting it to working over data and *not* function signatures (which drastically simplifies and slightly pivots the domain) -- and that puts us in a space that's... not... super well explored or well defined, relatively speaking, in popular literature.

So I'm generally +1 to trying to clarify that, but with the caveat of I think we'll absolutely land in "what are words" territory at some point and have to call it good enough.

File: schema/overview.md
Comment (warpfork) [reply schema/overview.md:47]: ... so, 'alias' might not be an ideal name for this feature, since it gives the association of multiple validity :thinking:

File: schema/overview.md
Comment (warpfork) [reply schema/overview.md:32]: hrm on second thought i'm not sure about this one, it feels a little clunky to me.  This is the overview page; do we wanna start using terms like that so early?  In subsequent pages, yes, but here I'm unsure.

maps and ~arrays~ lists are also kinds.  Maybe modifying the sentence prefacing this block of bulletpoints to say "[...] named types can be defined as any of the essential kinds"?

Comment (rvagg) [reply schema/overview.md:32]: maybe, or maybe you could make the assumption that they've read the "data model" page(s) as this is now 2 layers up from there
your call, not a big deal

File: schema/overview.md
Comment (rvagg) [reply schema/overview.md:47]: well the name "alias" made me assume it was an alternate form valid during deserialisation and that serialisation would always use the canonical name that's in the parent `type` section. So if it's a "**must** use alias", then yes, I think it's the wrong term. I don't have a suggestion for an alternative just at the moment though.

**Pull request closed without merging**
